### ISSUE TITLE
MLAG support?

### ISSUE THREAD

**Issue by gardar** [svlsResearch/ha-mikrotik#26]
Curious if anyone has tried using MLAG and ha-mikrotik together in RouterOS v7? MLAG does address some of the HA goals, but it looks like it could clash with the scripts (though I haven’t confirmed this myself). Any plans to add MLAG support to ha-mikrotik, or anyone who’s managed to use both without issues?"

**Comment by nathanfaber** [comment]
No plans for this. I use MLAG on my upstream Arista based switches with ha-mikrotik. My A Mikrotik going to arista A and B to Arista B. I am not using the MLAG RouterOS features though - they seem somewhat half baked and it isn't clear to me how to leverage this for ha-mikrotik.

What is the topology you are thinking?

**Comment by gardar** [comment]
I'm still planning to add a pair of CRS310-1G-5S-4S+IN routers at home, where I have a pair of Cisco Nexus switches serving as core switches. I initially thought it would be a good idea to connect the Nexus switches to the MikroTik devices in an active-active setup using MLAG, and possibly add VRRP on top for redundancy.

However, my WAN connection is provided via DHCP from my ISP, which means my WAN setup would likely need to operate in an active/passive mode, making VRRP probably unfeasible. While researching solutions for this DHCP-based WAN configuration, I came across this project, so I thought I could perhaps leverage it for the WAN connection along with MLAG on the other links.

**Comment by nathanfaber** [comment]
ha-mikrotik is half overkill for the home but a pair of Nexus switches is too, so I assume this is some home lab stuff. :)

It isn't clear to me how you are going to get rid of your upstream single point of failure in this setup? Do you have multiple ports to your upstream WAN device?

Is your goal to try be able to kill any single device (either a Nexus or a Mikrotik) and still keep everything running?

**Comment by gardar** [comment]
Haha, sure it's overkill, but that's what occurs when your work and hobbies are intertwined :)

The WAN is a single point of failure at the moment, I have a fiber media converter from the ISP with few ethernet ports so I can connect the mikrotik's to it, but I'll be replacing it with a SFP at some point and possibly using a Y or PON cable to be able to achieve redundancy.
I can get up to three public IP's from the ISP but they are all DHCP so I'm not sure if I'll be able to use it for VRRP on routeros.

The goal is indeed to be able to kill any single device without downtime and ideally have maximum throughput by leveraging the active/active nature of mlag / vpc.

**Comment by nathanfaber** [comment]
ha-mikrotik is inherently not active/active so I'm just not sure what role it would play here. The idea is to synchronize the config between an active/standby so you can do a failover without manually synchronizing things. You can achieve hardware redundancy but I am struggling to see how it can help if you are trying to achieve active/active. For example, to synchronize the config over to the standby, it literally reboots the device and restores a backup so there is no way active/active works.

**Comment by gardar** [comment]
I was hoping that it would perhaps be possible to mix it together, that is to choose specific parts of the config to sync and keep the WAN active/passive while also being able to use MLAG active/active for some of the interfaces.

**Comment by nathanfaber** [comment]
It simply is not possible with the way it works. The entire design evolves around making a full backup when it detects changes and then restoring it on the standby via a reboot.

**Comment by gardar** [comment]
But isn't there one interface that differs? The interface that's used to communicate between the active and standby.
Would it not be possible to just exclude the MLAG interfaces too?

**Comment by nathanfaber** [comment]
Yes, there is an interface used for direct communication between the pair. After the backup is restored post reboot, there is a bootstrapping script that quickly disables all the interfaces and then reconfigures that special interface to be distinct for the device. There is no getting around the reboot which makes active/active not make sense.
